<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>675</width>
    <height>796</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Sudoku Toy</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <layout class="QGridLayout" name="gridLayout_4">
    <item row="0" column="0">
     <layout class="QGridLayout" name="gridLayout_3">
      <item row="0" column="0">
       <widget class="QGroupBox" name="opts_gb">
        <property name="title">
         <string/>
        </property>
        <layout class="QGridLayout" name="gridLayout">
         <item row="0" column="0" colspan="5">
          <layout class="QGridLayout" name="gridLayout_2">
           <item row="0" column="1">
            <spacer name="verticalSpacer_2">
             <property name="orientation">
              <enum>Qt::Vertical</enum>
             </property>
             <property name="sizeHint" stdset="0">
              <size>
               <width>20</width>
               <height>40</height>
              </size>
             </property>
            </spacer>
           </item>
           <item row="1" column="0">
            <spacer name="horizontalSpacer_2">
             <property name="orientation">
              <enum>Qt::Horizontal</enum>
             </property>
             <property name="sizeHint" stdset="0">
              <size>
               <width>40</width>
               <height>20</height>
              </size>
             </property>
            </spacer>
           </item>
           <item row="1" column="1">
            <widget class="Grid" name="square_w" native="true">
             <property name="sizePolicy">
              <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
               <horstretch>0</horstretch>
               <verstretch>0</verstretch>
              </sizepolicy>
             </property>
             <property name="minimumSize">
              <size>
               <width>500</width>
               <height>500</height>
              </size>
             </property>
             <property name="maximumSize">
              <size>
               <width>500</width>
               <height>500</height>
              </size>
             </property>
             <property name="mouseTracking">
              <bool>true</bool>
             </property>
             <property name="accessibleDescription">
              <string extracomment="Grid widget"/>
             </property>
             <property name="autoFillBackground">
              <bool>false</bool>
             </property>
            </widget>
           </item>
           <item row="1" column="2">
            <spacer name="horizontalSpacer_3">
             <property name="orientation">
              <enum>Qt::Horizontal</enum>
             </property>
             <property name="sizeHint" stdset="0">
              <size>
               <width>40</width>
               <height>20</height>
              </size>
             </property>
            </spacer>
           </item>
          </layout>
         </item>
         <item row="1" column="0">
          <layout class="QHBoxLayout" name="horizontalLayout_6">
           <item>
            <widget class="QPushButton" name="new_pb">
             <property name="font">
              <font>
               <family>Cantarell</family>
               <bold>true</bold>
              </font>
             </property>
             <property name="text">
              <string>Clear</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QPushButton" name="solve_pb">
             <property name="font">
              <font>
               <bold>true</bold>
              </font>
             </property>
             <property name="text">
              <string>Solve</string>
             </property>
            </widget>
           </item>
          </layout>
         </item>
         <item row="1" column="1" colspan="3">
          <spacer name="horizontalSpacer">
           <property name="orientation">
            <enum>Qt::Horizontal</enum>
           </property>
           <property name="sizeHint" stdset="0">
            <size>
             <width>309</width>
             <height>20</height>
            </size>
           </property>
          </spacer>
         </item>
         <item row="1" column="4">
          <layout class="QHBoxLayout" name="horizontalLayout_5">
           <item>
            <widget class="QPushButton" name="undo_cb">
             <property name="font">
              <font>
               <bold>false</bold>
              </font>
             </property>
             <property name="text">
              <string>Undo</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QPushButton" name="redo_cb">
             <property name="font">
              <font>
               <bold>false</bold>
              </font>
             </property>
             <property name="text">
              <string>Redo</string>
             </property>
            </widget>
           </item>
          </layout>
         </item>
         <item row="2" column="0" colspan="2">
          <layout class="QHBoxLayout" name="horizontalLayout">
           <item>
            <widget class="QPushButton" name="pb_random">
             <property name="font">
              <font>
               <bold>true</bold>
              </font>
             </property>
             <property name="text">
              <string>Random</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QLabel" name="label">
             <property name="text">
              <string>Uncovered cells</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QLineEdit" name="le_nbcells">
             <property name="sizePolicy">
              <sizepolicy hsizetype="Preferred" vsizetype="Fixed">
               <horstretch>0</horstretch>
               <verstretch>0</verstretch>
              </sizepolicy>
             </property>
             <property name="minimumSize">
              <size>
               <width>40</width>
               <height>0</height>
              </size>
             </property>
             <property name="maximumSize">
              <size>
               <width>40</width>
               <height>16777214</height>
              </size>
             </property>
             <property name="text">
              <string>20</string>
             </property>
             <property name="alignment">
              <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
             </property>
            </widget>
           </item>
          </layout>
         </item>
         <item row="2" column="2">
          <spacer name="horizontalSpacer_4">
           <property name="orientation">
            <enum>Qt::Horizontal</enum>
           </property>
           <property name="sizeHint" stdset="0">
            <size>
             <width>153</width>
             <height>20</height>
            </size>
           </property>
          </spacer>
         </item>
         <item row="2" column="3" colspan="2">
          <layout class="QHBoxLayout" name="horizontalLayout_2">
           <item>
            <widget class="QLabel" name="lb_sols">
             <property name="text">
              <string>Max number of solutions</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QLineEdit" name="le_sols">
             <property name="sizePolicy">
              <sizepolicy hsizetype="Preferred" vsizetype="Fixed">
               <horstretch>0</horstretch>
               <verstretch>0</verstretch>
              </sizepolicy>
             </property>
             <property name="minimumSize">
              <size>
               <width>80</width>
               <height>0</height>
              </size>
             </property>
             <property name="maximumSize">
              <size>
               <width>80</width>
               <height>16777215</height>
              </size>
             </property>
             <property name="text">
              <string>1</string>
             </property>
             <property name="maxLength">
              <number>7</number>
             </property>
             <property name="alignment">
              <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
             </property>
            </widget>
           </item>
          </layout>
         </item>
         <item row="3" column="0" colspan="5">
          <widget class="QGroupBox" name="res_gb">
           <property name="sizePolicy">
            <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
             <horstretch>0</horstretch>
             <verstretch>0</verstretch>
            </sizepolicy>
           </property>
           <property name="title">
            <string/>
           </property>
           <layout class="QHBoxLayout" name="horizontalLayout_4">
            <item>
             <layout class="QHBoxLayout" name="horizontalLayout_3">
              <item>
               <widget class="QLabel" name="res_label">
                <property name="sizePolicy">
                 <sizepolicy hsizetype="MinimumExpanding" vsizetype="Preferred">
                  <horstretch>0</horstretch>
                  <verstretch>0</verstretch>
                 </sizepolicy>
                </property>
                <property name="font">
                 <font>
                  <italic>true</italic>
                 </font>
                </property>
                <property name="text">
                 <string/>
                </property>
                <property name="scaledContents">
                 <bool>true</bool>
                </property>
               </widget>
              </item>
              <item>
               <widget class="QPushButton" name="res_pb">
                <property name="sizePolicy">
                 <sizepolicy hsizetype="Preferred" vsizetype="Fixed">
                  <horstretch>0</horstretch>
                  <verstretch>0</verstretch>
                 </sizepolicy>
                </property>
                <property name="font">
                 <font>
                  <bold>false</bold>
                 </font>
                </property>
                <property name="text">
                 <string>Show another solution</string>
                </property>
               </widget>
              </item>
             </layout>
            </item>
           </layout>
          </widget>
         </item>
        </layout>
       </widget>
      </item>
     </layout>
    </item>
   </layout>
  </widget>
 </widget>
 <customwidgets>
  <customwidget>
   <class>Grid</class>
   <extends>QWidget</extends>
   <header location="global">src/grid.h</header>
   <container>1</container>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections/>
</ui>
